ui/src/message_popup/message_popup_page.jsx

```diff
@@ -90,10 +95,16 @@ export default React.createClass({
     this.setState({ shouldShowStudentCards: !this.state.shouldShowStudentCards });
   },
   
-  onHintsOptionToggled(){
-    this.setState({ allowedToToggleHint: !this.state.allowedToToggleHint });
+  onHelpToggled(event, value){
```

Comment:
This seems like it might be more natural to store the state as `scaffolding` and then the value can be `feedback | hints | none` rather than having two separate flags.  One benefit to that approach is that it makes it impossible to say `shouldShowFeedback: true, allowedToToggleHint: true`.